src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -116,18 +118,52 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             var unmatchedPathInOriginalTemplate = pathInExpandedTemplate[originalTemplatePath.Length..].TrimStart('.');
 
             // Compare the path of the expanded template with the path of the JToken found in the original template.
-            // If they match, or if the first unmatched property in the original is NOT a resources array,
+            // If they match or if the first unmatched property in the original is NOT a resources array
+            // and it's looking in a resources array and the first unmatched property in the original is an array index
             // the line number from the JToken of the original template can be returned directly.
             if (originalTemplatePath.Length == pathInExpandedTemplate.Length
-                || !unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase))
+                || (!unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase) 
+                    && !unmatchedPathInOriginalTemplate.StartsWith("[", StringComparison.OrdinalIgnoreCase)
+                    && !originalTemplatePath.EndsWith("resources", StringComparison.OrdinalIgnoreCase)))
             {
                 return (tokenFromOriginalTemplate as IJsonLineInfo)?.LineNumber ?? 0;
             }
 
-            // The first unmatched property must be a resources[] array, likely a subresource
-            // of a top-level resource.  This is possible if resources were copied into other
-            // resources as part of template expansion.
-            // TODO
+            string[] pathSegments = unmatchedPathInOriginalTemplate.Split('.');
+            string remainingPathAtResourceScope = string.Join('.', pathSegments[1..]);
+
+            var match2 = childResourceIndexInPath.Match(pathInExpandedTemplate);
```

Comment:
Why not reuse the `match` variable?

Reply:
It's inaccessible at this scope, so I renamed it to match.